electrum/util.py

```diff
@@ -681,7 +681,27 @@ def format_satoshis(
     if whitespaces:
         result += " " * (decimal_point - len(fract_part))
         result = " " * (15 - len(result)) + result
-    return result
+    dot = result.find('.')
```

Comment:
Note that, at this point, it might not be the `.` character that is used to separate the integer part and the fractional part:
it was replaced above with the locale-specific `DECIMAL_POINT`.

Reply:
Also note that the number was already split based on the decimal point into `integer_part` and `fract_part`; and then recombined into a single string.
Instead of splitting the number string and combining it again, you should move your logic further up and add the thousands separators there: perhaps right after where the extra zeros are added to the fractional part (line 678).